src/components/List/List.tsx

```diff
@@ -71,6 +81,23 @@ const List: FC<ListProps> = forwardRef(
       },
       [focusIndex]
     );
+
+    useEffect(() => {
+      if (!id) {
+        console.warn("List should have a valid id prop");
```

Comment:
does this mean the component can't function without it, if so, then it should be more than just a warning

Reply:
@shlomitc 
It can function, but the id(s) won't be unique, so it can lead to some troubles potentially
The logic is the following here:
- `List` doesn't have default id
- `ListItem` have default id, which is equal to `${listId}-item-${index}` - it's needed to set aria-activedescendants e.g.

So, if user don't pass `id` to `List`, he'll have following ListItem(s) ids:
`undefined-item-0`
`undefined-item-1`
...

And if user has more than 1 list rendered, elements ids won't be unique
So user should give an `id` to `List` and it all be good, but it's not a must
These ugly-looking ids (`undefined-item-0`) and a `console.warn` are supposed to help user to come to the right solution - give `List` an `id`

Tell me, if you have other suggestions on how to resolve it - we had the same issue with `Menu` in the past and eventually we came to this compromise solution with @orrgottlieb

Reply:
why do we need to set `id`s at all?
if we don't have to - just don't put it.

if we must for some reason, then I'd at least remove the `undefined` which looks like a mistake

Reply:
@shlomitc  For `ListItem` u mean?
We should always have them, cause we need to set some `ListItem` `id` to `List` `aria-activedescendant`